src/DiagonalHessianApproximation.jl

```diff
@@ -74,11 +81,31 @@ function push!(
   if B.psb
     q /= trA2
     B.d .+= q .* s .^ 2
-  else
+  elseif B.andrei
     sT_s = dot(s, s)
     q += sT_s
     q /= trA2
     B.d .+= q .* s .^ 2 .- 1
+  elseif B.wolk
+    sT_y0 = dot(s0, y0)
+    s02 = (si^2 for si ∈ s0)
+    sT_B_s0 = dot(s02, B.d)
+    sqrtd = (sqrt(di) for di ∈ B.d)
+    @assert all(x -> x > 0, B.d)
```

Comment:
The assertion comes too late. You've already computed the square root of each diagonal element!